### ISSUE TITLE
Add timeout to `TestStreamClient.get_event`

### ISSUE THREAD

**Issue by woile** [kpn/kstreams#151]
**Is your feature request related to a problem? Please describe.**
Yes, we thought our pipeline was hanging but instead, it was awaiting on the `get_event` when running the tests.

**Describe the solution you'd like**
I think the `get_event` should include a timeout by default. 
For most users, it should fail after some time.

**Describe alternatives you've considered**

**Additional context**

This should include a timeout by default, maybe 10 seconds (?)

With default timeout:

```python
client = TestStreamClient(stream_engine)

async with client:
    consumer_record = await client.get_event(topic_name=topic_name)
```

With custom timeout:

```python
from datetime import timedelta

client = TestStreamClient(stream_engine)

async with client:
    consumer_record = await client.get_event(topic_name=topic_name, timeout=timedelta(hours=1))
```

**Comment by woile** [comment]
Not enough quorum

**Comment by vervoorn** [comment]
Since this is for testing I do believe you should not wait. Can we make this or a similar method that just checks if there is an event? For example, a client.has_event(topic_name) or just a client.get_event(topic_name, wait_for_event=False) that returns immediately with None if there is no event.

**Comment by woile** [comment]
The question with those solutions is how do you account for time?
The "stream" runs in a task in background, in your test you cannot just fire an event and then check for "has_event" because you don't have guarantees that the "has_event" happens after your stream, this is specially true if your stream is slow

**Comment by marcosschroh** [comment]
I think having a `timeout` or checking whether  there is an event is a bad idea. The reasons  are:

1. Kafka: Consumers do not have a `timeout` and they do not ask whether there is an event in a topic
2. Having a `timeout` in the `get_event` can cause `false positives`, specially  if something previous to send an `event` is slower than your timeout
3. Having a method `has_event` can cause `false positives`. You could expect that there will be no events in a topic because the `send` was never called but actually your code could be slower than the test, then the test will pass but actually is should not.

Let's take a look an example:

```python
from .engine import engine


async def my_view(...):
    try:
        # do something here
        await engine.send("my-topic", ...)
    except Exception as e:
        return 
```
It is quite easy to test the success use case with the `TestStreamClient`. We  need the `test  client` here because we are "calling kafka", then we do not need to `mock`.

```python
import pytest

from .engine import engine
from .views import my_view

client = TestStreamClient(engine)


@pytest.mark.asyncio
async def test_my_view():
   
    async with client:
        await my_view()
        assert await client.get_event(topic_name="my-topic")
```

However, how to test the "failure" use case? If you use the `TestStreamClient` then it will `await` until there is an event, but never will be one. What you need to do is to test what your code is doing: the `send` is never called. If you assert that the `send` is  never called then it does not matter how slow or fast your function is, then any code change will be reflected in your `unittest`.

Now, let's go further. The `TestStreamClient` is mimicking `kafka`, what would happen if you decide to replace the `client` with an actual `kafka`? Then, the test can not be based on the a topic being empty or not.

In summary:

1. If you know that your code will produce an event (success use case) then use the  `TestStreamClient`
2. If you  know that an your code will not produce an event, then assert that the `send` was not called.
